### PR TITLE
feat: batching

### PR DESCRIPTION
**Pull request by BelfordZ**

### CODE REVIEW

**Comment by BelfordZ** [comment]
change name of `startBatch` -> `beginBatch`

**Comment by BelfordZ** [comment]
or change name of `endBatch` -> `stopBatch`

File: src/RequestManager.test.ts [hunk 1]
```diff
@@ -39,14 +39,97 @@ describe("client-js", () => {
...
+
+  it("can send a request and error", async () => {
     const transport = new EventEmitterTransport("foo://unique-uri");
     const c = new RequestManager([transport]);
     transport.onData = (fn) => {
       transport.connection.on("message", () => {
         fn(JSON.stringify({
           jsonrpc: "2.0",
-          id: 3,
+          id: 7,
```
Comment (shanejonas) [inline src/RequestManager.test.ts]: shouldnt this be 0?

**Pull request merged**